code/_onclick/hud/alert.dm

```diff
@@ -802,6 +804,166 @@ or shoot a gun to move around via Newton's 3rd Law of Motion."
 
 	ghost_owner.observer_view(target)
 
+/atom/movable/screen/alert/poll_alert
+	name = "Looking for candidates"
+	icon_state = "template"
+	timeout = 30 SECONDS
+	ghost_screentips = TRUE
+	var/show_time_left = FALSE // If true you need to call START_PROCESSING manually
+	var/image/time_left_overlay // The last image showing the time left
+	var/image/signed_up_overlay // image showing that you're signed up
+	var/image/stacks_overlay
+	var/image/candidates_num_overlay
+	var/image/role_name_overlay
+	var/datum/candidate_poll/poll // If set, on Click() it'll register the player as a candidate
+
+/atom/movable/screen/alert/poll_alert/Initialize(mapload)
+	. = ..()
+	register_context()
+
+/atom/movable/screen/alert/poll_alert/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = ..()
+	var/left_click_text
+	if(poll)
+		if(owner in poll.signed_up)
+			left_click_text = "Leave"
+		else
+			left_click_text = "Enter"
+		context[SCREENTIP_CONTEXT_LMB] = "[left_click_text] Poll"
+		if(poll.ignoring_category)
+			var/selected_never = FALSE
+			if(owner.ckey in GLOB.poll_ignore[poll.ignoring_category])
+				selected_never = TRUE
+			context[SCREENTIP_CONTEXT_ALT_LMB] = "[selected_never ? "Cancel " : ""]Never For This Round"
+		if(poll.jump_to_me && isobserver(owner))
+			context[SCREENTIP_CONTEXT_CTRL_LMB] = "Jump To"
+	return CONTEXTUAL_SCREENTIP_SET
+
+/atom/movable/screen/alert/poll_alert/process()
+	if(show_time_left)
+		var/timeleft = timeout - world.time
+		if(timeleft <= 0)
+			return PROCESS_KILL
+		cut_overlay(time_left_overlay)
+		var/obj/abstract_object = new
+		abstract_object.maptext = "<span style='font-family: \"Small Fonts\"; font-weight: bold; font-size: 8px; color: [(timeleft <= 10 SECONDS) ? "red" : "white"]; -dm-text-outline: 1px black'>[CEILING(timeleft / 10, 1)]</span>"
+		var/matrix/move_matrix = new
+		move_matrix.Translate(4, 16)
+		abstract_object.transform = move_matrix
+		time_left_overlay = image(abstract_object)
+		time_left_overlay.plane = plane
+		add_overlay(time_left_overlay)
+		qdel(abstract_object)
+	if(poll)
+		if(!role_name_overlay)
+			var/only_question = poll.role ? poll.role : "?"
+			var/obj/abstract_object = new
+			abstract_object.maptext = "<span style='font-family: \"Small Fonts\"; text-align: right; font-size: 7px; color: #B3E3FC; -dm-text-outline: 1px black'>[full_capitalize(only_question)]</span>"
+			abstract_object.maptext_width = 128
+			var/matrix/move_matrix = new
+			move_matrix.Translate(-128, 0)
+			abstract_object.transform = move_matrix
+			role_name_overlay = image(abstract_object)
+			role_name_overlay.plane = plane
+			add_overlay(role_name_overlay)
+			qdel(abstract_object)
```

Comment:
This seems *extremely* wasteful especially to be doing every process. Why are we going through this much rigamarole instead of keeping an object and settings it maptext? Why bother with images?